common/sampling.cpp

```diff
@@ -457,3 +466,20 @@ void llama_sampling_accept(
         llama_grammar_accept_token(ctx_main, ctx_sampling->grammar, id);
     }
 }
+
+bool is_english_token(const llama_context * ctx, llama_token token) {
+    const std::string token_str = llama_token_to_piece(ctx, token);
+    int english_char_count = 0;
+    bool has_angle_bracket = false;
+
+    for (char c : token_str) {
+        if (c >= 'a' && c <= 'z') {
```

Comment:
I agree that grammars are the proper way to address this task, but quick comment to note that if one is going to take this approach, then one should probably also check for upper-case characters as well.

```suggestion
        if ((c >= 'a' && c <= 'z') ||
            (c >= 'A' && c <= 'Z')) {
```